lang/cs/Org.Apache.REEF.Common/metrics/MetricsSystem/MetricsRecordBuilder.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using Org.Apache.REEF.Common.Metrics.Api;
+
+namespace Org.Apache.REEF.Common.Metrics.MetricsSystem
+{
+    /// <summary>
+    /// Default implementation of <see cref="IMetricsRecordBuilder"/>.
+    /// This class is used to collect records from <see cref="IMetricsSource"/> 
+    /// and convert them to <see cref="IMetricsRecord"/>. Mutable metrics from 
+    /// sources are observed and stores as immutable instances.
+    /// </summary>
+    internal sealed class MetricsRecordBuilder : IMetricsRecordBuilder
+    {
+        private readonly IList<IImmutableMetric> _metrics = new List<IImmutableMetric>();
+        private readonly IList<MetricsTag> _tags = new List<MetricsTag>();
+        private readonly IMetricsCollector _parentCollector;
```

Comment:
In current implementation it's unclear what is the role of `_parentCollector` and methods which return it - they are either not implemented or their return value is not used. Please either remove this field or add some functionality to use it